Shopping List/Controllers/ShoppingListController.swift

```diff
@@ -10,14 +10,30 @@ import Foundation
 
 class ShoppingListController {
 	
-	private(set) var list = [Item]()
 	private let notFirstTimeKey = "notFirstTime"
+	private(set) var list = [Item]()
+	
+	var filteredList: [(cartStatus: String, list: [Item])] {
+		var newList = [(cartStatus: String, list: [Item])]()
+		let added = list.filter({$0.isInCart})
+		let notAdded = list.filter({!$0.isInCart})
+		
+		if !added.isEmpty {
+			newList.append((cartStatus: "Added", list: added))
+		}
+		if !notAdded.isEmpty {
+			newList.append((cartStatus: "Not Added", list: notAdded))
+		}
+		
+		return newList
+	}
 	
```

Comment:
what you have obviously works, but this would be more clear and more efficient:

```
	var addedItems: [Item] {
		return list.filter({$0.isInCart}
	}

	var notAddedItems: [Item] {
		return list.filter({!$0.isInCart}
	}
```